learn/parse_http.go

```diff
@@ -443,6 +415,70 @@ func parseBody(contentType string, bodyStream io.Reader, statusCode int) (*pb.Da
 	return bodyData, nil
 }
 
+// When we can't parse the body, we will try to capture it as a raw primitive string and
+// indicate the parsing error in the body meta.
+func captureUnparsableBody(body memview.MemView, contentType string, statusCode int) *pb.Data {
+	mediaType, _, _ := mime.ParseMediaType(contentType)
+	bodyData := &pb.Data{
+		Value: newDataPrimitive(categorizeStringToPrimitive("Cannot parse body")),
+		Meta: newDataMetaHTTPMeta(&pb.HTTPMeta{
+			Location: &pb.HTTPMeta_Body{
+				Body: &pb.HTTPBody{
+					ContentType: getContentTypeFromMediaType(mediaType),
+					OtherType:   contentType,
+					Errors:      pb.HTTPBody_PARSING_ERROR,
+				},
+			},
+			ResponseCode: int32(statusCode),
+		}),
+	}
+
+	bodyStream := body.CreateReader()
```

Comment:
Non-blocking: It would be really nice if we could re-use the MemView all the way until the point we had to marshal this to a protobuf.  I don't think that is easy to do.  But, I count at least three copies here:

memview -> bodyBytes
bodyBytes -> bodyStr
primitive string -> copied into protobuf

I think we can eliminate one of those, using `strings.Builder`: https://pkg.go.dev/strings#Builder

```
buf := new(strings.Builder)
n, err := io.Copy(buf, body.CreateReader())
bodyStr := buf.String()
```

Reply:
Now that we are already creating a duplicate reader for the decoded stream, changed this to

```
buf := new(strings.Builder)
_, err := io.Copy(buf, bodyStream)
buf.String()
```